pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -887,31 +887,48 @@ func (oe *operationExecutor) generateVerifyControllerAttachedVolumeFunc(
 				volumeToMount.Pod.UID)
 		}
 
+		// This will write to node status that we're about to start using the volume.
+		// If we don't need the given volume, there might be a delay of upto 10s
+		// before we proclaim that we don't need it, because we *don't* synchronously
+		// delete the same volume from the node status map.
+		// TODO: consider fixing this post 1.3.
+		addVolumeNodeErr := actualStateOfWorld.MarkVolumeAsAttached(
```

Comment:
Marking a volume as attached in `VerifyControllerAttachedVolumeFunc` has two effects: 1) on the next node status update, the list of attached volumes is updated (which is what you want, ideally without the delay, but still what you want), and 2) marking the volume as attached which will cause the reconciler to stop trying to check if volume is attached. You are getting around 1 by forcing the MarkVolumeAsAttached method to update the node status. You are getting around 2 by by marking volume as detached if it doesn't exist in the controller's list. What you're missing is the errors that can happen in the remainder of this function--they will result in the reconciler falsely assuming the volume is attached. You can fix that by making sure you MarkVolumeAsDetached on each error, but this whole approach is *super* hacky